### PR TITLE
Respect blocks when automatically starting deploys

### PR DESCRIPTION
**Pull request by joeyAghion**
Our new concept of deploy strategies should respect our new concept of deploy blocks.

### CODE REVIEW

**Comment by joeyAghion** [comment]
Just added a commit that reorganized specs _only_. The small behavior change is reviewable in the first commit.

File: app/services/comparison_service.rb [hunk 1]
```diff
@@ -54,9 +54,11 @@ def refresh_comparisons
       new_snapshot = store_new_snapshot!(project, result, refreshed_at)
       clean_up_old_snapshots
     end
-    project.stages.select { |s| warrants_deploy?(s) }.each do |stage|
-      stage.deploy_strategies.each do |strategy|
-        DeployService.start(strategy) if strategy.automatic?
+    unless project.deploy_blocks.unresolved.any?
```
Comment (dleve123) [inline app/services/comparison_service.rb]: slight preference for `if project.deploy_blocks.unresolved.empty?` to keep things more positive, bit easier to understand IMO.

File: spec/features/organizations_spec.rb [hunk 2]
```diff
@@ -123,5 +122,24 @@
       expect(DeployService).not_to receive(:start)
       ComparisonService.new(project).refresh_comparisons
     end
+
+    it 'respects deploy blocks' do
```
Comment (dleve123) [inline spec/features/organizations_spec.rb]: thoughts on a spec description that better deblurs what "respects" means? How about "does not kick off a deploy if there are any unresolved deploy blocks"?

File: app/services/comparison_service.rb (hunk 1 above)
Comment (joeyAghion) [reply app/services/comparison_service.rb]: I actually had this as `if ... unresolved.none?` initially but the double negative felt bad. I think I'll go with your suggestion.

**Pull request merged**